Sources/Profile-Modify.php

```diff
@@ -1253,26 +1254,43 @@ function makeCustomFieldChanges($memID, $area, $sanitize = true, $returnErrors =
 					'member_affected' => $memID,
 				),
 			);
-			$changes[] = array(1, $row['col_name'], $value, $memID);
-			$user_profile[$memID]['options'][$row['col_name']] = $value;
+			if (empty($value))
+			{
+				$deletes = array('id_theme' => 1 , 'variable' => $row['col_name'], 'id_member' => $memID);
```

Comment:
I think this will only ever delete one row per save, won't it? This should probably use `$deletes[]` here, and then use a more advanced query below to delete all variables.

Reply:
Nope makeCustomFieldChanges() is called once for every field.

Reply:
Are you sure about that? The [query](https://github.com/SimpleMachines/SMF2.1/pull/4580/files#diff-73d40d702b0964f8eba5d0e9d7ea31b1R1159) retrieves all the fields in the profile area and then the [while](https://github.com/SimpleMachines/SMF2.1/pull/4580/files#diff-73d40d702b0964f8eba5d0e9d7ea31b1R1172) loops through them. Thats why `$changes[]` is used [here](https://github.com/SimpleMachines/SMF2.1/pull/4580/files#diff-73d40d702b0964f8eba5d0e9d7ea31b1R1264) rather than just `$changes`.

Reply:
Yes this query call only the >settings< for every fields.

Maybe because a field could got multipley values.
But this code delete the complet entry when value is empty.